### ISSUE TITLE
Holes in mesh(generative_B)

### ISSUE THREAD

**Issue by minjung98** [naver/posescript#26]
Hello. First, I'd like to thank you as your excellent research has been very helpful for my own research. I have a question: when I visualize the inference of generative_B, I can see holes here and there in the mesh. I'd like to ask why these holes are appearing and if it's normal for such holes to occur.
![image](https://github.com/user-attachments/assets/2f78e8c5-864a-4fec-b03a-d3187b632865)

**Comment by g-delmas** [comment]
Hi!
I remember having a similar problem at some point, but unfortunately I can't recall what the solution was. I think it simply disappeared when I re-ran the visualization app... does it occur systematically?
In any case, I don't think there is any problem with the generated joint rotations. It seems to be more of a rendering issue. You can try with another rendering pipeline, if you like!

**Comment by minjung98** [comment]
I have additional questions regarding generative_B:

1. Is the model at /posescript/experiments/b_gen_distilbert_dataPFAftPFH/seed1/checkpoint_best.pth a checkpoint that has completed both pretraining and fine-tuning?
2. For the joints generated by generative_B, is there a fixed camera position? I'm asking because it seems to change with each inference. I'm in the process of making fine adjustments to the pose, and it seems that the camera position is gradually changing to appear as if it's viewing from above, as shown below.

<img src="https://github.com/user-attachments/assets/b62dd6c2-67c7-4326-9599-ab5f6c8ff264" width="20%">

<img src="https://github.com/user-attachments/assets/ca979c55-b115-456f-9ff6-eb92dce46305" width="20%">

<img src="https://github.com/user-attachments/assets/1441e947-d99c-4240-824e-659c1e1c4d12" width="20%">

<img src="https://github.com/user-attachments/assets/ab9fe650-cf7e-4533-b20e-b5c91a97f7c9" width="20%">

<img src="https://github.com/user-attachments/assets/c3a03080-1c80-4ae0-85d4-746d41b0fa87" width="20%">

<img src="https://github.com/user-attachments/assets/92352765-6428-47cd-bd4f-f531247b03a0" width="20%">

<img src="https://github.com/user-attachments/assets/a595896a-98a8-4609-994f-22ac0ec3cfe4" width="20%">

<img src="https://github.com/user-attachments/assets/604d23a1-e9d6-4383-8f7e-09f7b78cac10" width="20%">

<img src="https://github.com/user-attachments/assets/be87c0c7-36ef-4587-a672-79eefe548a38" width="20%">

<img src="https://github.com/user-attachments/assets/9fcce9d0-e826-493f-bd57-4bb2e057a667" width="20%">

<img src="https://github.com/user-attachments/assets/0d16bfd7-70b3-47c3-b9cd-d26f3dbf5d03" width="20%">

<img src="https://github.com/user-attachments/assets/5a8ed99e-6ce4-474b-b2b7-1f4e2bdf91a9" width="20%">

<img src="https://github.com/user-attachments/assets/2b806f7e-f2a8-4898-9c77-a3b185f9b7c0" width="20%">

<img src="https://github.com/user-attachments/assets/89f7c23e-2873-46fb-aec8-05163bd7ad69" width="20%">

**Comment by minjung98** [comment]
> Hi! I remember having a similar problem at some point, but unfortunately I can't recall what the solution was. I think it simply disappeared when I re-ran the visualization app... does it occur systematically? In any case, I don't think there is any problem with the generated joint rotations. It seems to be more of a rendering issue. You can try with another rendering pipeline, if you like!

Actually, when visualizing the pictures I showed you earlier, I didn't use the visualization app. Instead, I used the checkpoint at "/posescript/experiments/b_gen_distilbert_dataPFAftPFH/seed1/checkpoint_best.pth" to perform inference and visualize the "gen_pose_data_samples". It doesn't seem to be an issue related to the streamlit app. From my various attempts, it seems that the degree of the holes changes slightly depending on the rotation...

**Comment by g-delmas** [comment]
Do you still get holes for the default pose? (with all joint rotations set to 0)

For the additional questions:

1. Yes, this checkpoint has completed both pretraining (_b_gen_distilbert_dataPFA_) and finetuning (+ _ftPFH_, with "ft" standing for "finetuning", and "PFH" for "PoseFix-H", while "PFA" stands for "PoseFix-A").
2.  The camera is kept fixed in the rendering process. Maybe variations in the global rotation of the body (ie. the root joint rotation) could make it look like the "camera" is moving across several renderings. However, I would find it surprising that the model iteratively modifies the root joint rotation in that way, since it has mainly been trained on poses facing front. The other training poses that do not face "front" are only rotated slightly to the left or the right, around the vertical axis. The training dataset does contain bodies hunched forward or bent backward, but still facing front. My best guess is that, in your own visualization pipeline, you are not applying the 90° base rotation (see [here](https://github.com/naver/posescript/blob/f5282b1115ab09a077ba50a04b230a939306c604/src/text2pose/utils_visu.py#L173)), meaning that you are actually seeing all the poses from the top instead of "from the front" (ie. the first pose is in fact a sitting pose).

**Comment by minjung98** [comment]
Thank you for your kind response. The camera position issue was due to a mistake I made while modifying the code. I haven't tried the default pose yet, but I'll try it later and let you know. This issue can be closed. Thank you very much for your help.
